CMakeLists.txt

```
@@ -21,3 +21,15 @@ add_subdirectory(subsys)
 add_subdirectory(modules)
 add_subdirectory(drivers)
 add_subdirectory(tests)
+
+if(CONFIG_SPM AND NOT CONFIG_IS_SPM)
+  include(${CMAKE_BINARY_DIR}/spm/shared_vars.cmake)
+  if(NOT "${CONFIG_FPU}" STREQUAL "${spm_CONFIG_FPU}")
+    add_custom_target(spm_fail
+		      COMMAND ${PYTHON_EXECUTABLE}
+		      -c \"print('Mismatch between fpu configurations in application and spm, CONFIG_FPU=${CONFIG_FPU}, spm_CONFIG_FPU=${spm_CONFIG_FPU}')\; exit(1)\"
```

Please no tabs in CMakeLists.txt files, just spaces.
```suggestion
      COMMAND ${PYTHON_EXECUTABLE}
        -c \"print('Mismatch between fpu configurations in application and spm, CONFIG_FPU=${CONFIG_FPU}, spm_CONFIG_FPU=${spm_CONFIG_FPU}')\; exit(1)\"
```

An extra question.
The old check was done in the `secure_services_ns.c` file, which is only included when `CONFIG_ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS=y`
https://github.com/nrfconnect/sdk-nrf/blob/c856a0476a167795fe6b42dbafc6fa0a24f45972/subsys/nonsecure/CMakeLists.txt#L7-L8

Should the build failure only happen when `CONFIG_ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS=y` or should we always fail when there is an FPU mismatch (current case in the PR now).